go/vt/vtgate/vcursor_impl.go

```diff
@@ -794,6 +794,11 @@ func (vc *vcursorImpl) SetPlannerVersion(v plancontext.PlannerVersion) {
 	vc.safeSession.GetOrCreateOptions().PlannerVersion = v
 }
 
+// SetConsolidator implements the SessionActions interface
+func (vc *vcursorImpl) SetConsolidator(consolidator querypb.ExecuteOptions_Consolidator) {
+	vc.safeSession.GetOrCreateOptions().Consolidator = consolidator
```

Comment:
I would suggest avoiding creating an Option if it does not already exist and the consolidator is unspecified.